easybot.js

```diff
@@ -0,0 +1,172 @@
+import { driver, settings } from '@rocket.chat/sdk';
+import dotenv from 'dotenv';
+import { config } from './config.js';
+import * as repository from './repository.js';
+dotenv.config();
+settings.username = process.env.USERNAME_ROCKETCHAT ;
+settings.password = process.env.PASSWORD;
+settings.host = process.env.HOST;
+let myuserid;
+
+const runbot = async () => {
+  await repository.createRedisClient();
+  await driver.connect();
+  myuserid = await driver.login();
+  await driver.subscribeToMessages();
+  console.log('subscribed');
+  driver.reactToMessages(processMessages);
+  console.log('connected and waiting for messages');
+}
+const processMessages = async (err, message) => {
+  if (!err) {
+    if (message.u._id === myuserid || (message.t && message.t==='au') || (!(await isUserPermitted( message.u.username)))) return;
+    if (message.msg.toLowerCase().startsWith(process.env.BOT_NAME)) {
+      await driver.sendToRoomId(await parseMessage(message.msg , message.u.username ), message.rid)
+    }
+  }
+}
+async function parseMessage(message , userName) {
+  let messageParts = message.split(';');
+  if (message === process.env.BOT_NAME) {
+    return userName + ",\n" + config.welcomeMsg;
+  }
+  const command=messageParts[1];
+  messageParts=messageParts.slice(2); // remove the cmd_bot and the command
+  switch (command) {
+    case "create_room": {
+      if(!validateArgumentsAmount(3,messageParts)){
+        return config.parametersMissingError;
+      }
+      return `${userName},\n*${(handleCreateRoomCommand(messageParts))}*\nFor: ${command}`;
+    }
+    case "set_user_active_status": {
+      if(!validateArgumentsAmount(2,messageParts)){
+        return config.parametersMissingError;
+      }
+      return `${userName},\n*${(await setUserActiveStatus(messageParts))}*\nFor: ${command}`;
+    }
+    case "update_roles_of_user": {  
+      if(!validateArgumentsAmount(3,messageParts)){
+        return config.parametersMissingError;
+      }
+      return `${userName},\n*${(await handleUpdateRolesCommand(messageParts))}*\nFor: ${command}`;
+    }
+    case "update_group_member": {  
+      if(!validateArgumentsAmount(3,messageParts)){
+        return config.parametersMissingError;
+      }
+      return `${userName},\n*${(await handleUpdateGroupMembersCommand(messageParts))}*\nFor: ${command}`;
+    }
+    case "send_message": {
+      if(!validateArgumentsAmount(2,messageParts)){
+        return config.parametersMissingError;
+      }
+      const messagePartsCopy=messageParts.join(';');
+      const text=messagePartsCopy.slice(0,(messagePartsCopy.lastIndexOf(';')));
```

Comment:
Ugly code and non readable!!
This needs to be done with a small function with meaning name, or with a comment.